pan-domain-auth-example/app/controllers/AdminController.scala

```diff
@@ -1,19 +1,24 @@
 package controllers
 
-import play.api.mvc.{ Action, Controller}
+import javax.inject.Inject
+
+import play.api.Configuration
+import play.api.mvc.{Action, Controller}
 import play.api.Play.configuration
 import play.api.Play.current
+import play.api.libs.ws.WSClient
 
 
-object AdminController extends Controller with ExampleAuthActions {
+class AdminController @Inject() (override val config: Configuration, override val wsClient: WSClient)
```

Comment:
I'd argue that it's worth converting to compile time dependency injection now at the earliest and least painful opportunity. Especially since you've got over the hump of the learning curve already. Assuming this project is going to continue to be developed on, the use of runtime dependency injection will only continue to grow. It's unlikely, but developers may also look as this project as a template in the future and decide to use runtime dependency injection as they have a working example. 

All in all less magic and craziness. That's my two cents. 😄